### ISSUE TITLE
Lists don't print count of current item and total count

### ISSUE THREAD

**Issue by joshferrell** [guidepup/virtual-screen-reader#40]
When using Voice Over for a `tablist` or a `radiogroup` each tab or radio item will be announced with the current index and total count of the items. In the case of virtual-screen-reader, this information is not printed. This is a useful test to confirm that when provided a list of radio options in a radiogroup or tabs in a tablist that they are being rendered within the group.

## Example HTML
```html
<div role="tablist" aria-orientation="horizontal">
  <button
    type="button"
    role="tab"
    aria-selected="true"
    aria-controls="tab-:r6::Tab1-panel"
    id="tab-:r6::Tab1-trigger"
    tabindex="0"
  >
    Tab1
  </button>
  <button
    type="button"
    role="tab"
    aria-selected="false"
    aria-controls="tab-:r6::Tab2-panel"
    id="tab-:r6::Tab2-trigger"
    tabindex="-1"
  >
    Tab2
  </button>
</div>
<div role="tabpanel" aria-labelledby="tab-:r6::Tab1-trigger" tabindex="0">
  <div>Tab1 Content</div>
</div>
```

## Example Test

```tsx
const { container } = render(<TestComponent />)
console.log(container.innerHTML)
await virtual.start({ container })
while ((await virtual.lastSpokenPhrase() !== 'end of tabpanel, Tab1')) {
  await virtual.next()
}

// this expect passes
expect(await virtual.spokenPhraseLog()).toEqual([
  "tablist, orientated horizontally",
  "tab, Tab1, selected",
  "tab, Tab2, not selected",
  "end of tablist, orientated horizontally",
  "tabpanel, Tab1",
  "Tab1 Content",
  "end of tabpanel, Tab1"
])

// this expect fails
expect(await virtual.spokenPhraseLog()).toEqual([
  "tablist, orientated horizontally",
  "tab, Tab1, selected, 1 of 2",
  "tab, Tab2, not selected, 1 of 2",
  "end of tablist, orientated horizontally",
  "tabpanel, Tab1",
  "Tab1 Content",
  "end of tabpanel, Tab1"
])
await virtual.stop()
```

**Comment by cmorten** [comment]
Hey @joshferrell 👋 

Thanks for raising! I’m a little undecided with this one(!) though my gut feeling is that it is a good thing to do.

Till now this project has largely been spec driven (though I’m sure I’ve likely been biased by what certain SRs do), and afaik there is no commentary or requirement (implicit or explicit) in specs that state this is something SRs are expected to do.

On the other hand, to be actually _useful_ as a mirror of SR behaviour (though care needed to never claim to match real SRs) perhaps using precedent as a feature driver is just as valuable as the specs? I can see how adding the index in this scenario would assist in making the snapshot less ambiguous. I suspect it would be even more useful in scenarios such as nested lists where asserting hierarchy has been successfully achieved is useful.

Which brings us to decided for which implicit or explicit roles should there be the expectation that the (1) total count is announced on the parent (if we want this as well) and (2) the index within the containing parent is announced on the child(ren)?

**Comment by cmorten** [comment]
Relevant spec sections:

- https://www.w3.org/TR/wai-aria-1.2/#aria-level
- https://www.w3.org/TR/wai-aria-1.2/#aria-posinset
- https://www.w3.org/TR/wai-aria-1.2/#aria-setsize
- https://www.w3.org/TR/core-aam-1.2/#mapping_additional_position

There are no expectations defined in https://a11ysupport.io/ from what I can see.

aria-at has some assertions indicating that the position and set size announcement should be expected:

- https://github.com/w3c/aria-at/blob/master/tests/tabs-manual-activation/data/tests.csv
- TBD

**Comment by cmorten** [comment]
Released in `0.21.0`
